cpp/CMakeLists.txt

```diff
@@ -35,16 +37,34 @@ endif(CMAKE_COMPILER_IS_GNUCXX)
 set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -D_GLIBCXX_USE_CXX11_ABI=0")
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -D_GLIBCXX_USE_CXX11_ABI=0")
 
-#set(CMAKE_CUDA_FLAGS "${CMAKE_CUDA_FLAGS} -gencode=arch=compute_60,code=sm_60 -gencode=arch=compute_61,code=sm_61")
-set(CMAKE_CUDA_FLAGS "${CMAKE_CUDA_FLAGS} -gencode=arch=compute_60,code=sm_60")
-set(CMAKE_CUDA_FLAGS "${CMAKE_CUDA_FLAGS} -gencode=arch=compute_70,code=sm_70 -gencode=arch=compute_70,code=compute_70")
+# e.g. cmake -DGPU_COMPUTE_CAPABILITY=60 ... to support GPU Compute Capability 6.0
+if(NOT GPU_COMPUTE_CAPABILITY)
+    message(AUTHOR_WARNING "NVIDIA GPU Compute Capability is not defined, using defult NVIDIA GPU compute capability version 7.0")
+    set(GPU_COMPUTE_CAPABILITY "70")
```

Comment:
Looks like the defaults are compute 60 and 70, why are we changing the defaults to only 70 here now?

Reply:
Furthermore, this doesn't look like it can be easily extended to allow someone to build for multiple compute capabilities, i.e. if they wanted to distribute a single shared library across a heterogeneous cluster.

Reply:
This is not how we define CMake variables. Try something like:
```
set(GPU_COMPUTE_CAPABILIUTY CACHE STRING "Description of this variable and its syntax goes here")
```
However, you might want the cached variable to be a "forced value", with it's default being the empty string. Then, if there's no forcing, you should use compute capability detection.